static/js/unread.js

```diff
@@ -707,4 +748,13 @@ export function initialize() {
     for (const message_id of unread_msgs.mentions) {
         unread_messages.add(message_id);
     }
+
+    $(document).on("home_view_loaded.zulip", () => {
+        // Once we are finished loading all the unread messages,
+        // update the unread_mention_topics and re-render recent topics.
```

Comment:
I don't see why we'd need to wait for the home view to load? It seems like we should have all the mentions in the `unread_msgs` structure...

Reply:
It only has the message ids.

Reply:
Ahh. We have the location of all of them via the other data structures, just not split out. So I think we could replace this with a loop like this one and remove the need to do anything later, which could lead to weird races:
```
    get_streams_with_unread_mentions() {                                                                 
        const streams_with_mentions = new Set();                                                         
        // Collect the set of streams containing at least one mention.                                   
        // We can do this efficiently, since unread_mentions_counter                                     
        // contains all unread message IDs, and we use stream_ids as                                     
        // bucket keys in our outer bucketer.                                                            
                                                                                                         
        for (const message_id of unread_mentions_counter) {                                              
            const stream_id = this.bucketer.reverse_lookup.get(message_id);                              
            const per_stream_bucketer = this.bucketer.get_bucket(stream_id);
            const topic_match = per_stream_bucketer.reverse_lookup.get(message_id);
            ....
        }                                                                                                
                                                                                                         
        return streams_with_mentions;                                                                    
    }                                                                                                    
```



with some error handling added

Reply:
`UnreadTopicCounter` is also populated after we receive messages from the server, not before.

Reply:
@timabbott Fixed, thanks for the help!